espressif/idf-eclipse-plugin
# IEP-370: NVS partition editor

## Description

Check more about NVS partition editor utility from esp-idf, check this here https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/storage/nvs_partition_gen.html

This PR includes the preparation for refactoring PartitionTableDataUtil class to be extended from AbstactTableDataService

Fixes # ([IEP-370](https://jira.espressif.com:8443/browse/IEP-370))

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How has this been tested?

Test 1:
- Right click on the project in the project explorer -> click ESP-IDF: NVS Table Editor -> The table with default values is provided
- Click Save and Quit -> nvs.csv should be created in project directory

Test 2:
- Try to put incorrect values into the field (More info [here](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/storage/nvs_partition_gen.html)) and Save and then generate nvs.bin 
- In this case, you should see an error notification with a clear error message as to what is wrong.

Test 3:
- Input correct values -> Save -> Generate Partition
- In this case, you should see an informational message about the successful saving and creation of partitions.
- nvs.bin should be present in project directory

Test 4: 
- Generate partition with and without encryption
- Generate key and select the encryption key


**Test Configuration**:
* ESP-IDF Version:
* OS (Windows,Linux and macOS):

## Dependent components impacted by this PR:

- NVS Table Editor

## Checklist
